test/wasmfs/wasmfs_fetch.c

```diff
@@ -150,13 +150,42 @@ void test_small_reads() {
   assert(close(fd) == 0);
 }
 
+void test_nonexistent() {
+  printf("Running %s...\n", __FUNCTION__);
+
+  const char* dir_path = "/subdir";
+  char url[200];
+  snprintf(url, sizeof(url), "%s%s", url_orig, dir_path);
+
+  backend_t backend = wasmfs_get_backend_by_path(url);
```

Comment:
All the other tests seem to call `wasmfs_create_fetch_backend`. Is there a reason not to do that in this new test?

Reply:
Yes, I initially used `wasmfs_create_fetch_backend` as in other tests. But this requires modifying test_browser.py. I had to increase pool size from 5 to 6 and also add '-sINITIAL_MEMORY=32MB' flag to make it work because when running tests I was getting the following error: 
`Cannot enlarge memory arrays to size 17858560 bytes (OOM). Either (1) compile with -sINITIAL_MEMORY=X with X higher than the current value 16777216, (2) compile with -sALLOW_MEMORY_GROWTH which allows increasing the size at runtime, or (3) if you want malloc to return NULL (0) instead of this abort, compile with -sABORTING_MALLOC=0`.

So to avoid modifications to test_browser.py I decided to reuse existing backend. But if you think that we need to use  `wasmfs_create_fetch_backend` here then I can do this along with changes to test_browser.py.

Reply:
I see, good points. Yes, maybe for testing a nonexistent file this is simpler as it is.